### DISCUSSION TITLE
Superscript & Subscript Text

### DISCUSSION THREAD

**Discussion by girlpunk** [QuestPDF/QuestPDF#104] [Ideas]
The new text rendering capabilities are a great improvement, one other useful feature would be superscript and subscript text. I currently have this working through a rather hacky solution, though obviously this isn't ideal 

```csharp
// Size is internal, so need to use reflection to access it
var sizeProperty = typeof(TextStyle).GetProperty("Size", BindingFlags.Instance | BindingFlags.NonPublic);
var currentSize = (float) sizeProperty.GetValue(_currentStyle);

// These two can be adjusted as nessasary to alter the offset and size of the text
var newSize = currentSize / 1.6f;
var newPadding = currentSize / 2.5f;

_currentStyle = _currentStyle.Size(newSize);
textDescriptor.Element()
    .PaddingBottom(newPadding) // Use -newPadding for subscript
    .Text(innerTextDescriptor => innerTextDescriptor.span("superscript text", _currentStyle));
```

**Comment by MarcinZiabek** [comment]
This is a great idea! I will put superscript and subscript on the roadmap. At this point, I have no idea how difficult it is. I expect that information about proper font size and positioning (when using superscript or subscript) are somewhere within font metadata. Once I extract this information, I should be able to render text in the proper size and position. 

This is very similar to how underline or strike font metrics are used.

**Comment by MarcinZiabek** [comment]
Hello 😁 I have just released the QuestPDF 2022.5 version that supports this feature: subscript and superscript. [Please take a look here.](https://www.questpdf.com/documentation/api-reference.html#subscript-and-superscript)

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
